pkg/ccl/storageccl/export_target.go

```diff
@@ -0,0 +1,413 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Cockroach Community Licence (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/ccl/LICENSE
+
+package storageccl
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/pkg/errors"
+
+	gcs "cloud.google.com/go/storage"
+	"github.com/rlmcpherson/s3gof3r"
+)
+
+const (
+	// S3AccessKeyParam is the query parameter for access_key in an s3 URI.
+	S3AccessKeyParam = "AWS_ACCESS_KEY_ID"
+	// S3SecretParam is the query parameter for the 'secret' in an s3 URI .
+	S3SecretParam = "AWS_SECRET_ACCESS_KEY"
+)
+
+// ExportStorageConfFromURI generates an ExportStorage config from a URI string.
+func ExportStorageConfFromURI(path string) (roachpb.ExportStorage, error) {
+	conf := roachpb.ExportStorage{}
+	uri, err := url.Parse(path)
+	if err != nil {
+		return conf, err
+	}
+	switch uri.Scheme {
```

Comment:
Are we concerned about capitalization here? What if a user specifies `GS://stuff`? Should we use `strings.ToLower(uri.Scheme)`?